### ISSUE TITLE
.htaccess files not being generated?

### ISSUE THREAD

**Issue by mikerockett** [piwik/piwik#10420]
I've just started  using Piwik on a cloud server, and it appears that the installation did not create the needed `.htaccess` files. As a result, all directories are indexable.

The public folder has all the necessary permissions where my user (owner) and the `apache` user (group) can write.

Is there any way I can have these files generated, and why did they not generate in the first place? Better yet, why are these files not included by default anyway? I'm sure a simple `.htaccess` in the root would have sufficed, instead of one for each directory (I think that's what the generator does?)...

**Edit:** I also think the security module should warn users when these files do not exist.

**Update:** I have now generated these files by calling the relevant method from index.php (deleted now as this is a once-off thing).

```php
Piwik\Plugins\Installation\ServerFilesGenerator::createHtAccessFiles();
```

I also noted from a [GH search](/piwik/piwik/search?utf8=%E2%9C%93&q=createHtAccessFiles) that this method is only called when running updates, but never at install-time. Is this an oversight?

**Comment by mattab** [comment]
Hi @mikerockett 
We have fixed this issue recently in PR: https://github.com/piwik/piwik/pull/10414 which will be merged for in 2.16.3. Feel free to test the PR. Thanks for the report!

**Comment by mikerockett** [comment]
:+1: Thanks
